Dockerfile

```diff
@@ -380,7 +380,8 @@ RUN apt-get install -y curl
 RUN mkdir -p /dart-protobuf
 ARG PROTOC_GEN_DART_VERSION
 RUN curl -sSL https://api.github.com/repos/google/protobuf.dart/tarball/protoc_plugin-${PROTOC_GEN_DART_VERSION} | tar xz --strip 1 -C /dart-protobuf
-WORKDIR /dart-protobuf/protoc_plugin 
+WORKDIR /dart-protobuf/protoc_plugin
+ARG PUB_HOSTED_URL=https://pub.flutter-io.cn
```

Comment:
Could you add a comment briefly explaining what this is and what it's for?

Reply:
Yep, was just going to ask you about this. This is the bug that has been blocking all of our CI runs for the last month, just now getting a bit of time to look at it.

The official host for the Dart package manager is https://pub.dartlang.org/, which can be searched at https://dart.dev/. Several users on [corporate networks](https://github.com/dart-lang/pub/issues/3317#issuecomment-1150027103), in countries with [poor connectivity](https://stackoverflow.com/questions/71265574/flutter-pub-publish-returns-error-69-connection-closed-before-full-header-was-r) and [internet restrictions](https://stackoverflow.com/a/71331866) report difficulty connecting to this host, and other users report the same difficulty when running in CI and/or emulation. It seems the remote server has a strict security configuration that is dropping connections in some cases if the wrong [user agent](https://github.com/dart-lang/sdk/issues/48450#issuecomment-1050414712) is used.

The easiest solution is to use a mirror of the package host with a different configuration that doesn't drop the connection for whatever reason. These tend to be hosted in China, due to the enormous popularity of Android there, and the difficulty in accessing outside internet, particularly unpredictable speed throttling which is super annoying when trying to download large packages. The URL is recommended in the [Flutter docs](https://docs.flutter.dev/community/china#configuring-flutter-to-use-a-mirror-site).

I don't think there is any security risk because the packages are still verified for authenticity and integrity with a signature and checksum, like with any other package manager. But if you prefer, we can disable ARM build for Dart instead to avoid depending on the non-canonical package manager URL.